connman/gsupplicant/supplicant.c

```diff
@@ -2135,9 +2235,22 @@ static void interface_property(const char *key, DBusMessageIter *iter,
 	} else if (g_strcmp0(key, "Networks") == 0) {
 		supplicant_dbus_array_foreach(iter, interface_network_added,
 								interface);
-	} else
+	} else if (g_strcmp0(key, "DisconnectReason") == 0) {
+		int reason_code;
+		if (dbus_message_iter_get_arg_type(iter) != DBUS_TYPE_INVALID) {
```

Comment:
the usual: variable type is wrong and type test is interesting, but fingers crossed...

Reply:
Actually, we don't even use this internal g_supplicant.  We use gsupplicant from [libgsupplicant](https://github.com/sailfishos/libgsupplicant). We can ignore this, we are not going to switch back to this apparently and there are more fixes on this in the upstream when we upgrade further.